Sources/SwiftParser/Directives.swift

```diff
@@ -256,14 +259,34 @@ extension Parser {
       args = nil
     }
     let (unexpectedBeforeRParen, rparen) = self.expect(.rightParen)
+    let unexpectedAfterRightParen = self.consumeRemainingTokenOnLine()
+
     return RawPoundSourceLocationSyntax(
       poundSourceLocation: line,
       unexpectedBeforeLParen,
       leftParen: lparen,
       args: args,
       unexpectedBeforeRParen,
       rightParen: rparen,
+      unexpectedAfterRightParen,
       arena: self.arena
     )
   }
+
+  /// Consumes remaining token on the line and returns a ``RawUnexpectedNodesSyntax``
+  /// if there is any tokens conrumed.
+  private mutating func consumeRemainingTokenOnLine() -> RawUnexpectedNodesSyntax? {
+    var rawUnexpectedNodesSyntax: RawUnexpectedNodesSyntax?
+    if !self.currentToken.isAtStartOfLine {
+      var unexpectedTokens = [RawTokenSyntax]()
+      var loopProgress = LoopProgressCondition()
+      while !self.at(.eof), !currentToken.isAtStartOfLine, loopProgress.evaluate(self.currentToken) {
+        unexpectedTokens += [self.consumeAnyToken()]
+      }
+
+      rawUnexpectedNodesSyntax = RawUnexpectedNodesSyntax(unexpectedTokens, arena: self.arena)
+    }
+
+    return rawUnexpectedNodesSyntax
```

Comment:
Just an idea: We could remove the nesting level by early-exiting if `self.currentToken.isAtStartOfLine`.

```suggestion
    guard !self.currentToken.isAtStartOfLine else {
      return nil
    }
    var unexpectedTokens = [RawTokenSyntax]()
    var loopProgress = LoopProgressCondition()
    while !self.at(.eof), !currentToken.isAtStartOfLine, loopProgress.evaluate(self.currentToken) {
      unexpectedTokens += [self.consumeAnyToken()]
    }

    return RawUnexpectedNodesSyntax(unexpectedTokens, arena: self.arena)
```